### DISCUSSION TITLE
how to: getText() for nodes/paragraphs and complex input rules

### DISCUSSION THREAD

**Discussion by Kukuschi** [ueberdosis/tiptap#2346] [General Questions]
Hello,
i want to add more complex input rules to my editor e.g. to automate the setting of quotation marks. (» and «), change minus to ndash under certain conditions etc. . 
I did this before in python + pyside with the QT text editor widget*: 

get text of current paragraph
on typing " do backward search from cursor position for » and «
If the closest quotation mark to cursor position is » change " to «, if it is « or not existing change it to »

I am not sure how to do it in TipTap (and if it is possible)
Can i get just the text of the paragraph the cursor is in with getText() in tiptap? (The rest can be done with quite simple JS logic i guess?)  and inject the changes only to the paragraph again via setText() somehow? (It seems to only get the Text of the hole editor content (?) and that might get a bit too heavy on very long texts).

*Working on a novel structuring and writing software and switched from Python + Pyside to Vue 3 (i am fairly new to js, css and html, tho and am not a very experienced coder in general) as it seemed more easy to achive the Gui style i had in mind (and because of some other limitations of QT i did not like).

**Comment by philippkuehn** [comment]
Use something like `editor.state.selection.$from.parentNode.textContent` or a helper method like [textBetween](https://prosemirror.net/docs/ref/#model.Fragment.textBetween).
